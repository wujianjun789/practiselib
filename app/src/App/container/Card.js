/**
 * create by Azrael on 2017/7/4
 */
import React from 'react';
import {Link} from 'react-router';

export const Card = ({_key, title, link}) => {
    return <Link to={link}>
        <div className={`card card-${_key}`}>
            <div className="ico"><span className={`icon icon-${_key}`}></span></div>
            <div className="title">{title}</div>
        </div>
    </Link>
}