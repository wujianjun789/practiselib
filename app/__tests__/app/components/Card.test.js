/**
 * created by Azrael on 2017/7/5
 */
import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Card from '../../../src/App/container/Card';

describe('<Card />', () => {
    let data = {
        key: 'asset',
        title: '资产管理',
        link: '/asssetMessage'
    }
    it('render normal', () => {
        const card = shallow(<Card _key={data.key} title={data.title} link={data.link}/>);
        let link = card.find('Link');
        expect(link.length).toBe(1);
        expect(link.prop('to')).toBe(data.link);
        
        let cmp = card.find('.card.card-asset');
        expect(cmp.length).toBe(1);

        let icon = cmp.find('.icon.icon-asset');
        expect(icon.length).toBe(1);

        let tit = cmp.find('.title');
        expect(tit.length).toBe(1);
        expect(tit.text()).toBe(data.title);
    });

    it('snapshot', () => {
        const card = renderer.create(<Card _key={data.key} title={data.title} link={data.link} />);
        let tree = card.toJSON();
        expect(tree).toMatchSnapshot();
    });
});