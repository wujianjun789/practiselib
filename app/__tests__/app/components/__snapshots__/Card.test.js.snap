exports[`<Card /> snapshot 1`] = `
<a
  onClick={[Function]}
  style={Object {}}>
  <div
    className="card card-asset">
    <div
      className="ico">
      <span
        className="icon icon-asset" />
    </div>
    <div
      className="title">
      资产管理
    </div>
  </div>
</a>
`;
